### DISCUSSION TITLE
Seeking Advice for Efficient Data Storage and Optimization in a HDD-Based Cluster

### DISCUSSION THREAD

**Discussion by okgame** [minio/minio#17504] [Q&A] [answered]
Hello,

I am currently planning to construct a PB cluster using HDD.

The original data that I intend to store varies in size from 200MB to 1GB. Each day, we predict that approximately 1% will be written to and around 10% of the total storage capacity of the cluster will be read. This reading process will be conducted in segments, through byte range requests.

Our cluster nodes are interconnected via a WAN network. In order to optimize bandwidth usage for internal queries among nodes, I've encountered an issue when reading large files: even if the client machine only requests a small portion of the file, the node processing the request still consumes a significant amount of line In, exceeding the volume of data requested by the client.

Thus, I am seeking advice on the following two options:

1. Should I upload and download the original files, ranging from 200MB to 1GB in size, to the minio cluster?
2. Or should I break down the original files into smaller segments, approximately 10-20MB each?

In case I choose to divide the files, how small should the segments be in MB to achieve a balance between the processing capacity of the HDD and the optimization of network bandwidth among nodes?

I look forward to your advice,

Sincerely, thank you.

**Comment by klauspost** [comment, marked as answer]
@okgame We do not provide community setup/optimization guidance. Please reach out to `hello@min.io` and we can guide you directly.
